Lombiq.BaseTheme/Middlewares/RemoveBootstrapMiddleware.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Lombiq.BaseTheme.Constants;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using OrchardCore.DisplayManagement.Manifest;
+using OrchardCore.Environment.Extensions;
+using OrchardCore.ResourceManagement;
+using OrchardCore.Themes.Services;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Lombiq.BaseTheme.Middlewares;
+
+/// <summary>
+/// Removes the built-in Bootstrap resource if the currently selected theme uses Lombiq.BaseTheme as its base theme.
+/// Themes derived from Lombiq.BaseTheme use Bootstrap from NPM as it's compiled into their site stylesheet. So this
+/// duplicate resource is not needed and can cause problems if not removed. This situation can arise when a module
+/// (such as Lombiq.DataTables) depends on Bootstrap and doesn't explicitly depend on Lombiq.BaseTheme so the built-in
+/// resource would injected if this middleware didn't remove it.
+/// </summary>
+public class RemoveBootstrapMiddleware
+{
+    private static readonly object _lock = new();
+    private static readonly string[] _resourceTypes = { "stylesheet", "script" };
+
+    private readonly RequestDelegate _next;
+
+    private static bool _done;
```

Comment:
This static variable is problematic for multi-tenancy so I've removed it, the rest is fine. I verified it like this?
1. Place a breakpoint at the `lock` below.
2. Set up the site
3. Create a new tenant with the url prefix `test`.
4. Go to _~/Lombiq.DataTables.Samples/Sample/DataTableTagHelper_, the breakpoint triggers once.
5. Go to _/test/Lombiq.DataTables.Samples/Sample/DataTableTagHelper_, the breakpoint triggers again.
6. Reloading either page doesn't trigger the breakpoint.

It looks like singleton service exist per-tenant, so it's fine for one to use the theme and others not. (I've tested it as well.) Also this means if you want to change back to a different theme that uses the stock Bootstrap you just have to reload the tenant. I've added some notes about that.

Reply:
Having to reload a tenant is fine, since that happens already when you enable/disable a theme (or did you only try setting an already enabled theme as the current one?).

Reply:
Okay, but if you want to swap between two themes that are already enabled that does not reload the tenant, that's why I added the warning. I did this:
1. Go to ~/Lombiq.DataTables.Samples/Sample/DataTableTagHelper.
2. Bootstrap CSS is missing as expected
3. Go to Admin > Design > Themes and change to Blog theme.
4. Go to the DataTables page: ❌Boostrap CSS is still missing.
5. Go to Admin > Configuration > Tenants.
6. Click <kbd>Reload</kdb> button.
7. Go to the DataTables page: ✔️ Boostrap CSS is now correctly present.

Reply:
Yeah, that's what I was referring to as "setting an already enabled theme as the current one". OK then.